saleor/graphql/product/types.py

```diff
@@ -58,6 +72,18 @@ def resolve_availability(self, info):
             self, context.discounts, context.currency)
         return ProductAvailability(**availability._asdict())
 
+    def resolve_attributes(self, info):
+        return resolve_instance_attributes(instance=self, info=info)
```

Comment:
I suppose `instance` refers to the fact that both Product and Variant have the field `attributes`, so we have a single method to get the attributes in both cases. If we renamed this field to something else this function would break. Instead, it could be more generic and take the attributes dict:
```
def resolve_attributes(self, info):
    return resolve_attributes(self.attributes.items())
```